src/api/RebalancingSetIssuanceAPI.ts

```diff
@@ -0,0 +1,228 @@
+      txOpts,
+    );
+
+    return this.rebalancingSetIssuanceModule.issueRebalancingSetWrappingEther(
+      rebalancingSetAddress,
+      rebalancingSetQuantity,
+      keepChangeInVault,
+      txOpts,
+    );
+  }
+
+  /**
+   * Redeems a Rebalancing Set into the base SetToken. Then the base SetToken is redeemed, and its components
+   * are sent to the caller or transferred to the caller in the Vault depending on the keepChangeInVault argument.
+   *
+   * @param  rebalancingSetAddress    Address of the rebalancing Set to redeem
+   * @param  rebalancingSetQuantity   Quantity of the rebalancing Set to redeem
+   * @param  keepChangeInVault        Boolean signifying whether excess base SetToken is transfered to the user
+   *                                    or is left in the vault
+   * @param  txOpts                   The options for executing the transaction
+   */
+  public async redeemRebalancingSet(
+    rebalancingSetAddress: Address,
+    rebalancingSetQuantity: BigNumber,
+    keepChangeInVault: boolean,
+    txOpts: Tx
+  ): Promise<string> {
+    await this.assert.issuance.assertRedeem(
+      rebalancingSetAddress,
+      rebalancingSetQuantity,
+      txOpts.from,
+    );
+
+    return this.rebalancingSetIssuanceModule.redeemRebalancingSet(
+      rebalancingSetAddress,
+      rebalancingSetQuantity,
+      keepChangeInVault,
+      txOpts,
+    );
+  }
+
+  /**
+   * Redeems a Rebalancing Set into the base SetToken. Then the base Set is redeemed, and its components
+   * are sent to the caller. Any wrapped Ether is unwrapped and transferred to the caller.
+   *
+   * @param  rebalancingSetAddress    Address of the rebalancing Set to redeem
+   * @param  rebalancingSetQuantity   Quantity of the rebalancing Set to redeem
+   * @param  keepChangeInVault        Boolean signifying whether excess base SetToken is transfered to the user
+   *                                    or is left in the vault
+   * @param  txOpts                    The options for executing the transaction
+   */
+  public async redeemRebalancingSetUnwrappingEther(
+    rebalancingSetAddress: Address,
+    rebalancingSetQuantity: BigNumber,
+    keepChangeInVault: boolean,
+    txOpts: Tx
+  ): Promise<string> {
+    await this.assertRedeemRebalancingSetUnwrappingEther(
+      rebalancingSetAddress,
+      rebalancingSetQuantity,
+      txOpts,
+    );
+
+    return this.rebalancingSetIssuanceModule.redeemRebalancingSetUnwrappingEther(
+      rebalancingSetAddress,
+      rebalancingSetQuantity,
+      keepChangeInVault,
+      txOpts,
+    );
+  }
+
+  /* ============ Private Assertions ============ */
+
+  private async assertIssueRebalancingSetWrappingEther(
+    rebalancingSetAddress: Address,
+    rebalancingSetQuantity: BigNumber,
+    txOpts: Tx,
+  ) {
+    this.assert.common.isNotUndefined(txOpts.value, exchangeIssuanceErrors.ETHER_VALUE_NOT_UNDEFINED());
+
+    await this.assert.issuance.assertRebalancingSetTokenIssueWrappingEther(
```

Comment:
Is there a reason why we ferry all of these parameters to another class to do the assertion instead of doing it here? Somewhere along the way we added too much abstraction. To my knowledge, only things that are reusable should in there but the name is `assertRebalancingSetTokenIssueWrappingEther` which is really specific.